File: apps/admin/src/types/payments.ts
export type ChargebackStatus = "received" | "under_review" | "accepted" | "fighting" | "won" | "lost";
export type P2PStatus = "pending" | "confirmed" | "rejected" | "sent" | "completed";
export type P2PType = "deposit" | "withdrawal";
export type P2PMethod = "papara" | "bank_transfer" | "crypto_p2p";
export type ReconciliationStatus = "pending" | "resolved" | "investigating";

export interface Chargeback {
  id: string;
  player_id: string;
  player_email: string;
  transaction_id: string;
  amount: string;
  currency: string;
  gateway: string;
  gateway_cb_id: string | null;
  reason_code: string | null;
  reason_text: string | null;
  status: ChargebackStatus;
  received_at: string;
  deadline_at: string | null;
  resolved_at: string | null;
  assigned_to: string | null;
  assigned_to_name: string | null;
  fight_evidence: ChargebackEvidence[];
  notes: string | null;
  days_to_deadline: number | null;
}

export interface ChargebackEvidence {
  type: string;
  url: string;
  uploaded_at: string;
}

export interface ChargebackQueueFilters {
  status?: ChargebackStatus;
  gateway?: string;
  assigned_to?: string;
  page?: number;
  page_size?: number;
}

export interface ChargebackActionPayload {
  action: "accept" | "fight" | "assign";
  evidence?: ChargebackEvidence[];
  notes?: string;
}

export interface BalanceSheet {
  as_of: string;
  liabilities: {
    player_balances: string;
    bonus_balances: string;
    pending_withdrawals: string;
    total: string;
  };
  assets: {
    gateways: GatewayBalance[];
    crypto_hot: CryptoBalance[];
    crypto_cold: CryptoBalance[];
    bank_account: string;
    total: string;
  };
  coverage_ratio: number;
}

export interface GatewayBalance {
  name: string;
  balance: string;
  currency: string;
}

export interface CryptoBalance {
  coin: string;
  amount: string;
  usd_equivalent: string;
}

export interface CryptoWallet {
  id: string;
  coin: string;
  wallet_type: "hot" | "cold";
  balance: string;
  address: string;
  daily_withdrawal_avg: string;
  threshold_amount: string;
  is_low: boolean;
  pending_deposits: number;
  pending_withdrawals: number;
  last_updated: string;
}

export interface P2PTransaction {
  id: string;
  player_id: string;
  player_email: string;
  type: P2PType;
  amount: string;
  currency: string;
  method: P2PMethod;
  status: P2PStatus;
  receipt_url: string | null;
  confirmed_by: string | null;
  confirmed_by_name: string | null;
  confirmed_at: string | null;
  notes: string | null;
  created_at: string;
  hours_waiting: number;
}

export interface P2PQueueFilters {
  status?: P2PStatus;
  type?: P2PType;
  method?: P2PMethod;
  page?: number;
  page_size?: number;
}

export interface P2PActionPayload {
  action: "confirm" | "reject" | "mark_sent";
  notes?: string;
}

export interface PaymentMethodConfig {
  id: string;
  country_code: string;
  method: string;
  gateway: string;
  enabled_deposit: boolean;
  enabled_withdrawal: boolean;
  min_deposit: string;
  max_deposit: string | null;
  min_withdrawal: string;
  max_withdrawal: string | null;
  fee_percent: string;
  fee_fixed: string;
  priority: number;
  temporary_disabled_until: string | null;
}

export interface ReconciliationRecord {
  id: string;
  recon_date: string;
  gateway: string;
  expected_balance: string;
  actual_balance: string;
  difference: string;
  pending_tx_count: number;
  failed_callbacks: number;
  chargeback_amount: string;
  status: ReconciliationStatus;
  notes: string | null;
}

export interface ChargebackStats {
  total_this_month: number;
  amount_this_month: string;
  cb_rate_pct: number;
  fight_win_rate_pct: number;
}

File: services/go/admin-bff/internal/models/payments.go
package models

import (
	"encoding/json"
	"time"
)

type Chargeback struct {
	ID               string          `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	PlayerID         int64           `gorm:"not null;index" json:"player_id"`
	PlayerEmail      string          `gorm:"type:varchar(255);not null" json:"player_email"`
	TransactionID    string          `gorm:"type:varchar(100);not null" json:"transaction_id"`
	Amount           string          `gorm:"type:numeric(18,2);not null" json:"amount"`
	Currency         string          `gorm:"type:char(3);not null;default:'USD'" json:"currency"`
	Gateway          string          `gorm:"type:varchar(100);not null" json:"gateway"`
	GatewayCbID      *string         `gorm:"type:varchar(255)" json:"gateway_cb_id,omitempty"`
	ReasonCode       *string         `gorm:"type:varchar(50)" json:"reason_code,omitempty"`
	ReasonText       *string         `gorm:"type:varchar(500)" json:"reason_text,omitempty"`
	Status           string          `gorm:"type:chargeback_status;not null;default:'received'" json:"status"`
	ReceivedAt       time.Time       `gorm:"not null;default:now()" json:"received_at"`
	DeadlineAt       *time.Time      `json:"deadline_at,omitempty"`
	ResolvedAt       *time.Time      `json:"resolved_at,omitempty"`
	AssignedTo       *string         `gorm:"type:varchar(36)" json:"assigned_to,omitempty"`
	AssignedToName   *string         `gorm:"type:varchar(255)" json:"assigned_to_name,omitempty"`
	FightEvidence    json.RawMessage `gorm:"type:jsonb;default:'[]'" json:"fight_evidence,omitempty"`
	Notes            *string          `gorm:"type:text" json:"notes,omitempty"`
	CreatedAt        time.Time        `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt        time.Time        `gorm:"not null;default:now()" json:"updated_at"`
}

type CryptoWallet struct {
	ID                   string    `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	Coin                 string    `gorm:"type:varchar(20);not null" json:"coin"`
	WalletType           string    `gorm:"type:varchar(10);not null;check:wallet_type IN ('hot','cold')" json:"wallet_type"`
	Balance              string    `gorm:"type:numeric(18,8);not null;default:0" json:"balance"`
	Address              string    `gorm:"type:varchar(255);not null" json:"address"`
	DailyWithdrawalAvg   string    `gorm:"type:numeric(18,8);not null;default:0" json:"daily_withdrawal_avg"`
	ThresholdAmount      string    `gorm:"type:numeric(18,8);not null;default:0" json:"threshold_amount"`
	PendingDeposits      int       `gorm:"not null;default:0" json:"pending_deposits"`
	PendingWithdrawals   int       `gorm:"not null;default:0" json:"pending_withdrawals"`
	LastUpdated          time.Time `gorm:"not null;default:now()" json:"last_updated"`
	CreatedAt            time.Time `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt            time.Time `gorm:"not null;default:now()" json:"updated_at"`
}

type P2PTransaction struct {
	ID                string     `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	PlayerID          int64      `gorm:"not null;index" json:"player_id"`
	PlayerEmail       string     `gorm:"type:varchar(255);not null" json:"player_email"`
	Type              string     `gorm:"type:p2p_type;not null" json:"type"`
	Amount            string     `gorm:"type:numeric(18,2);not null" json:"amount"`
	Currency          string     `gorm:"type:char(3);not null;default:'TRY'" json:"currency"`
	Method            string     `gorm:"type:p2p_method;not null" json:"method"`
	Status            string     `gorm:"type:p2p_status;not null;default:'pending'" json:"status"`
	ReceiptURL        *string    `gorm:"type:varchar(512)" json:"receipt_url,omitempty"`
	ConfirmedBy       *string    `gorm:"type:varchar(36)" json:"confirmed_by,omitempty"`
	ConfirmedByName   *string    `gorm:"type:varchar(255)" json:"confirmed_by_name,omitempty"`
	ConfirmedAt       *time.Time `json:"confirmed_at,omitempty"`
	Notes             *string    `gorm:"type:text" json:"notes,omitempty"`
	CreatedAt         time.Time  `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt         time.Time  `gorm:"not null;default:now()" json:"updated_at"`
}

type ReconciliationRecord struct {
	ID                string    `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	ReconDate         time.Time `gorm:"type:date;not null" json:"recon_date"`
	Gateway           string    `gorm:"type:varchar(100);not null" json:"gateway"`
	ExpectedBalance   string    `gorm:"type:numeric(18,2);not null;default:0" json:"expected_balance"`
	ActualBalance     string    `gorm:"type:numeric(18,2);not null;default:0" json:"actual_balance"`
	Difference        string    `gorm:"type:numeric(18,2);not null;default:0" json:"difference"`
	PendingTxCount    int       `gorm:"not null;default:0" json:"pending_tx_count"`
	FailedCallbacks   int       `gorm:"not null;default:0" json:"failed_callbacks"`
	ChargebackAmount  string    `gorm:"type:numeric(18,2);not null;default:0" json:"chargeback_amount"`
	Status            string    `gorm:"type:reconciliation_status;not null;default:'pending'" json:"status"`
	Notes             *string    `gorm:"type:text" json:"notes,omitempty"`
	CreatedAt         time.Time  `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt         time.Time  `gorm:"not null;default:now()" json:"updated_at"`
}

type PaymentMethodConfig struct {
	ID                     string     `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	CountryCode            string     `gorm:"type:char(2);not null" json:"country_code"`
	Method                 string     `gorm:"type:varchar(50);not null" json:"method"`
	Gateway                string     `gorm:"type:varchar(100);not null" json:"gateway"`
	EnabledDeposit         bool       `gorm:"not null;default:true" json:"enabled_deposit"`
	EnabledWithdrawal      bool       `gorm:"not null;default:true" json:"enabled_withdrawal"`
	MinDeposit             string     `gorm:"type:numeric(18,2);not null;default:0" json:"min_deposit"`
	MaxDeposit             *string    `gorm:"type:numeric(18,2)" json:"max_deposit,omitempty"`
	MinWithdrawal          string     `gorm:"type:numeric(18,2);not null;default:0" json:"min_withdrawal"`
	MaxWithdrawal          *string    `gorm:"type:numeric(18,2)" json:"max_withdrawal,omitempty"`
	FeePercent             string     `gorm:"type:numeric(5,4);not null;default:0" json:"fee_percent"`
	FeeFixed               string     `gorm:"type:numeric(18,2);not null;default:0" json:"fee_fixed"`
	Priority               int        `gorm:"not null;default:0" json:"priority"`
	TemporaryDisabledUntil *time.Time `json:"temporary_disabled_until,omitempty"`
	CreatedAt              time.Time  `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt              time.Time  `gorm:"not null;default:now()" json:"updated_at"`
}
